internal/config/instance.go

```diff
@@ -169,7 +176,31 @@ func Get() (*Instance, error) {
 	return defaultConfig, nil
 }
 
-// Set sets a value at the given key
+// SetUnsafe updates a configuration key and writes the update configuration file to disk.
+// IMPORTANT: This function does not ensure any inter process synchronization!
+// The caller needs to explicitly guard this function by calling
+// GetLock()/ReleaseLock().
+// Example usage:
+//    cfg.GetLock()
+//    defer cfg.ReleaseLock()
+//    cfg.ReloadUnsafe()
+//    slice, _ := cfg.GetStringSlice("key")
+//    slice = append(slice, "value")
+//    cfg.SetUnsafe("key", slice)
+// Note, how in this example it is crucial to synchronize the appending to the slice variable.
```

Comment:
Nit: In the docstring for `ReloadUnsafe` it says to see this docstring for when and how to use this function, however, this docstring is explaining how and not when. I can see from how you are using it in this PR that you want to hold the config lock until the service has started so no other services can start so we may want to add some more detail to this already long docstring. Or remove the reference in the `ReloadUnsafe` docstring